examples/pokemon-service/src/authz.rs

```diff
@@ -0,0 +1,176 @@
+//! This file showcases a rather minimal model plugin that is agnostic over the operation that it
+//! is applied to.
+//!
+//! It is interesting because it is not trivial to figure out how to write one. As the
+//! documentation for [`aws_smithy_http_server::plugin::ModelMarker`] calls out, most model
+//! plugins' implementation are _operation-specific_, which are simpler.
+
+use std::{marker::PhantomData, pin::Pin};
+
+use aws_smithy_http_server::{
+    body::BoxBody,
+    operation::OperationShape,
+    plugin::{ModelMarker, Plugin},
+};
+use pokemon_service_server_sdk::server::response::IntoResponse;
+use tower::Service;
+
+pub struct AuthorizationPlugin {
+    // Private so that users are forced to use the `new` constructor.
+    _private: (),
+}
+
+impl AuthorizationPlugin {
+    pub fn new() -> Self {
+        Self { _private: () }
+    }
+}
+
+impl<Ser, Op, T> Plugin<Ser, Op, T> for AuthorizationPlugin {
+    type Output = AuthorizeService<Op, T>;
+
+    fn apply(&self, input: T) -> Self::Output {
+        AuthorizeService {
+            inner: input,
```

Comment:
I added a comment.

Reply:
I understand the confusion though. We should probably change the `Plugin` definition to take in `inner` instead of `input`: https://docs.rs/aws-smithy-http-server/latest/aws_smithy_http_server/plugin/trait.Plugin.html

Reply:
On second thought, `input` is probably a better name than `inner`. There's nothing preventing a plugin from returning a `Service` that doesn't wrap `T`.